cfn-resources/global-cluster-config/cmd/resource/resource.go

```diff
@@ -84,31 +79,24 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 }
 
 func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
-	setup() // logger setup
+	setup()
 
-	// Validate required fields in the request
-	if modelValidation := validateModel(RequiredFields, currentModel); modelValidation != nil {
-		return *modelValidation, nil
+	if errValidation := validateModel(RequiredFields, currentModel); errValidation != nil {
+		return *errValidation, nil
 	}
 
-	// Create MongoDb Atlas Client using keys
-	if currentModel.Profile == nil || *currentModel.Profile == "" {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
-	}
-	client, pe := util.NewMongoDBClient(req, currentModel.Profile)
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	client, pe := util.NewAtlasClient(&req, currentModel.Profile)
 	if pe != nil {
-		_, _ = logger.Warnf("CreateMongoDBClient error: %v", *pe)
 		return *pe, nil
 	}
 
-	// Check if  already exist
-	if !isExist(*client, currentModel) {
-		return progressevents.GetFailedEventByCode("Resource Not Found", cloudformation.HandlerErrorCodeNotFound), nil
+	if !isExist(client, currentModel) {
+		return progressevent.GetFailedEventByCode("Resource Not Found", cloudformation.HandlerErrorCodeNotFound), nil
 	}
-	// method call to read configuration
-	config, event, err := ReadConfig(*client, currentModel)
+
+	config, event, err := ReadConfig(client, currentModel)
```

Comment:
isExist calls ReadConfig so we're calling the server twice to get the same info when one call would be enough. (i know this is how it was before).   One easy way to fix it is to pass "config" (server response) to isExist

Reply:
you're right. I updated the READ operation to not use `isExist`. We still need to call `ReadConfig` as part of `isExist` as part of the delete function